runtime/src/accounts_index.rs

```diff
@@ -831,13 +836,16 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         self.purge_secondary_indexes_by_inner_key(pubkey, Some(&purged_slots), account_indexes);
     }
 
-    pub fn clean_rooted_entries(
+    pub fn clean_rooted_entries<F>(
         &self,
         pubkey: &Pubkey,
         reclaims: &mut SlotList<T>,
         max_clean_root: Option<Slot>,
         account_indexes: &HashSet<AccountIndex>,
-    ) {
+        should_purge: F,
```

Comment:
nits: fyi, you could write like this (this should be identical in terms of actual binary; shorter and concise), not needing the `where` part:

```rust
      should_purge: impl Fn(&T) -> bool,
```

Reply:
also, considering its sole usage, I think it's fine to be specific in naming so, how about this?

`is_cache_and_not_purge` or like that.

Reply:
especially, this `should_purge` is confusing with `this `should_purge`: https://github.com/solana-labs/solana/pull/13140/files#r553427624

Reply:
updated to `is_cached`!